ekf: refactoring unused gps functions up to current code standards

niewim19 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes shortly -->
As work has begun to fully implement GPS into AHRS (delay before new barometer arrive), some work need to be done on the state of EKF filter so that next changes are not implemented into code with flawed structure. For that the following PR consists of commits that:
 - restore functions regarding GPS usage to meet current state of repository
 - restructure code (`measurement.h` separation)
 - fix functions inputs to use proper structures and not loads of variables

**Note:** as EKF variable naming was not up to standards it is corrected in this PR, but the order of changes is:
 1) restructure
 2) fix naming
So if there is some variable naming consideration it may be fixed in the following commits of this PR

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
JIRA: PP-107

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

HBuczynski commented on ekf/kalman_implem.h:
```diff
@@ -142,57 +143,16 @@ typedef struct {
-	geodetic_t gpsRefGeodetic;
-} kalman_calib_t;
-
 int verbose;
```
Try to not use global variable.

niewim19 replied:
Implemented as separate commit that changes init values handling from using global variables to passing a pointer

HBuczynski commented on ekf/kalman_implem.h:
```diff
@@ -142,57 +143,16 @@ typedef struct {
-/* CALIBRATIONS */
-/* TODO: move to separate, calibration-related header */
 
 void read_config(void);
```
Add some prefix to function.

niewim19 replied:
Change added as separate commit

HBuczynski commented on ekf/measurement.c:
```diff
@@ -53,26 +54,26 @@ float acc_calib1[12] = {
 
 
-vec_t geo2ecef(float lat, float lon, float h)
+static vec_t meas_geo2ecef(meas_geodetic_t *pos)
```
```suggestion
static vec_t meas_geo2ecef(const meas_geodetic_t *pos)
```

HBuczynski replied:
The better approach would be to pass `vec_t` as a second argument to avoid copying within returning data.

HBuczynski reviewed (changes requested):
Some remarks:
1. There is a prefix `meas` in `measurement.h` and measurement.c`. I would change the name of this unit to `meas`.
2. I am not sure whether there is a need to make a granulation to `kalman_implem` nad `kalman_core`. You can merge it to one unit, e.x: `core` or `ekf`.

niewim19 commented on ekf/measurement.h:
```diff
@@ -0,0 +1,85 @@
+/* CALIBRATION INITIALIZERS */
+
+/* obtain current IMU calibration parameters */
+void meas_imuCalib(void);
+
+/* obtain current barometer calibration parameters */
+void meas_baroCalib(void);
+
+/* obtain current gps calibration parameters */
+void meas_gpsCalib(void);
+
+
+/* CALIBRATION GETTERS */
+
+/* Return pointer to full calibration data */
+const meas_calib_t *meas_calibGet(void);
+
+/* returns the calibration pressure in Pascals */
+float meas_calibPressGet(void);
+
+
+/* MEASUREMENT ACQUISITION */
+
+/* Returns prepared IMU data in SI units */
+int meas_imuGet(vec_t *accels, vec_t *gyros, vec_t *mags, uint64_t *timestamp);
+
+/* Returns prepared barometer data in SI units */
+int meas_baroGet(float *pressure, float *temperature, uint64_t *dtBaroUs);
+
+/* Returns prepared GPS data in SI units */
+int meas_gpsGet(vec_t *enu, vec_t *enu_speed, float *hdop);
+
+#endif
```
Change added in separate commit

niewim19 commented:
Changes made since last review:
 - implemented function naming review remarks that are resolved (mainly as separate commits at the head of branch)
 - switched to passing positions (actual/reference/ecef/enu) as pointers
 - revised usage of globals (currently only one matrix in kalman_core.c is left as global) but its enough for this PR
 - changed handling of initialization parameters to kalman filter:
   - dropped usage of global array `init_values`
   - implemented one array that holds initialization coefficients. Pointer to this structure is passed across `ekf` module
   - config read is not changed as it needs a parser usage and it would be too much for this PR

HBuczynski commented on ekf/kalman_implem.c:
```diff
@@ -22,14 +22,15 @@
 
 /* NOTE: must be kept in the same order as 'config_names' */
-kalman_init_t init_values = {
+const kalman_init_t initTemplate = {
```
```suggestion
static const kalman_init_t initTemplate = {
```

HBuczynski commented on ekf/kalman_implem.c:
```diff
@@ -62,7 +63,7 @@ kalman_init_t init_values = {
 
 /* NOTE: must be kept in the same order as 'init_values' */
-char *config_names[] = {
+const char *config_names[] = {
```
```suggestion
static const char *configNames[] = {
```

HBuczynski commented on ekf/meas.c:
```diff
@@ -39,11 +40,11 @@
 
 /* accelerometer calibration data */
-float acc_calib1[12] = {
+static float acc_calib1[12] = {
```
In the name of array, there is a magic number. Can you change it ?
```suggestion
static const float accCalib1[12] = {
```

niewim19 replied:
Used two macros: 
```c
/* different accelerometer biases sizes */
#define ACC_BIAS_SIZE    3
#define ACC_NONORTHO_SIZE 9
```
Precompiled array will be dropped as soon as config parser will be used

HBuczynski commented on ekf/kalman_implem.c:
```diff
@@ -85,73 +88,72 @@ void read_config(void)
+	cov->data[cov->cols * IMZ + IMZ] = inits->P_merr * inits->P_merr;
 }
 
 vec_t last_a = { 0 };
```
It is only used in `calcStateEstimation`.
Please add it as `static` variable inside this function.

niewim19 replied:
Thanks! I missed that global variable.
